### ISSUE TITLE
Technique for validating DataGrid record no longer working

### ISSUE THREAD

**Issue by CJMarinacci** [stsrki/Blazorise#2166]
I used to be able to validate a single new or updated record in a DataGrid by wrapping the entire DataGrid in a Validation control and then manually calling Validation.Validate() when the record was inserted or updated. That no longer works because Validation controls now have a private inputComponent field that is (as far as I can tell) only initialized from components derived from BaseInputComponent<TValue>. Trying to use any other component will throw an exception (or validate using the wrong component) when you try to validate.

Is there a recommended way of doing what I used to be able to do (validate a single record in a DataGrid on inserting or updating)? I was able to somewhat recreate the previous behavior by calling Validation.NotifyValidationStatusChanged() on the DataGrid's Validation component, but is there a better way?

Thanks,

Chris

**Comment by stsrki** [comment]
I don't quite understand why it is not working for you. Can you post a sample code of how you use DataGrid with Validation?

**Comment by CJMarinacci** [comment]
The following is the basic structure. "CommandColumn" is a custom component with command buttons. "FeedbackComponent" is a custom component that shows feedback text.

```
<Validations>
    <Validation>
        <DataGrid>
            <DataGridDateColumn/>
            <DataGridDateColumn/>
            <DataGridColumn/>
            <DataGridColumn/>
            <DataGridColumn/>
            <CommandColumn/>
        </DataGrid>
    </Validation>
    <FeedbackComponent/>
</Validations>
```

I was calling the "Validate" method of the Validation component, and a NullReferenceException occurred because the "inputComponent" field of the Validation component was null. This was working in a previous version of Blazorise -- I'm not sure which one, because I hadn't tested this particular part of the code in a while.

If you need more information, I can try digging into the code some more.

Thanks,

Chris

**Comment by stsrki** [comment]
Hmm, that's a weird way to validate DataGrid? You're wrapping entire grids around a single validation component 🤔

Why are you using it like that and what actually are you validating that way? DataGrid has its own validation methods for the editing fields. So why not use it?

**Comment by CJMarinacci** [comment]
I needed to compare the values of two of the fields of the record that were being edited, as opposed to validating each of the fields separately.

**Comment by stsrki** [comment]
Would it work for you to validate on RowUpdating maybe? Recently I have recently added ValidationsSummaryErrors to the DataGrid so that custom errors could be used.

For reference: https://github.com/stsrki/Blazorise/pull/2107

**Comment by CJMarinacci** [comment]
Yes, I'm validating on RowInserting and RowUpdating, and I changed my code to update Validation (which updates FeedbackComponent) if there is a validation error (which works). 

I can't find DataGrid.ValidationsSummaryErrors in 0.9.3.4.

**Comment by stsrki** [comment]
I tend to forget not everyone is on preview :)

It is released as part of 0.9.4 preview 1, on MyGet https://github.com/stsrki/Blazorise#try-preview, if you're willing to try.

**Comment by CJMarinacci** [comment]
What I have right now is working, so I'll wait until it's out of preview. Thanks!
